torch/_refs/__init__.py

```diff
@@ -441,6 +443,14 @@ def _convert(x):
     return tuple(map(lambda x: _convert(x), args))
 
 
+def _unwrap_cpu_scalars(a, b):
```

Comment:
There's a million funny interactions here, and yeah if this unwrapping is done before type promotion it'll through it off (but it's done after, so that's fine). The reason for unwrapping user-provided tensors is that they stop beind 0d tensors after forced broadcasting #76871

Reply:
So we really want to keep the `elif` here? meaning that one will become a Number and the other will remain a 0D Tensor?

Reply:
As it stands today (because prim implementation calls into `torch.add` and not into `torch.ops.aten.add` ) it doesn't matter, because python argparser will rewrap them again. But yeah, it's an interesting question, and if prim starts calling into aten, like it should, it would matter.

Reply:
I think we should handle this by folding "scalar unwrapping" like this into a new type promotion wrapper I'm working on now, but no worries if this has an opportunity to land first